client/src/view/components/Header/Nav/Search/Search.jsx

```diff
@@ -0,0 +1,138 @@
+import React, {useEffect, useMemo, useState} from 'react';
+
+import useAsyncError from "../../../../hooks/useAsyncError";
+
+import {useDispatch, useSelector} from "react-redux";
+import {searchResultActions} from "../../../../../redux/features/searchResult";
+import {isAnyDropdownOpenActions, isAnyDropdownOpenSelectors} from "../../../../../redux/features/dropdown";
+
+import {useHistory} from "react-router-dom";
+import {productRequests} from "../../../../../api/server";
+
+import Toast from "../../../Toast/Toast";
+import HeaderDropdown from "../../HeaderDropdown/HeaderDropdown";
+import Button from "../../../Button/Button";
+import {Box, InputAdornment, TextField} from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+
+import search from "../icons/search.svg";
+import styles from "./Search.module.scss";
+import useWindowSize from "../../../../hooks/useWindowSize";
+import constants from "../../../../constants";
+import NavItemsStyles from "../NavItems/NavItems.module.scss";
+
+Search.propTypes = {};
+
+function Search(props) {
+    const [searchValue, setSearchValue] = useState("");
+    const [searchResult, setSearchResult] = useState([]);
+    const [isLoaded, setLoaded] = useState(false);
+    const [isDropdownActive, setActiveDropdown] = useState(false);
+
+    const isAnyDropdownOpen = useSelector(isAnyDropdownOpenSelectors.getIsAnyDropdownOpen);
+    const dispatch = useDispatch()
+    const throwAsyncError = useAsyncError();
+    const history = useHistory();
+
+    const [isDesktop, setIsDesktop] = useState()
+    const sizes = useWindowSize();
+
+    useEffect(() => {
```

Comment:
This can be simplified to `setIsDesktop(sizes.width >= constants.WINDOW_DESKTOP_SIZE )`